frontend/projects/ui/src/app/pages/apps-routes/app-show/pipes/to-buttons.pipe.ts

```diff
@@ -112,23 +115,67 @@ export class ToButtonsPipe implements PipeTransform {
     await modal.present()
   }
 
-  private viewInMarketplaceButton(pkg: PackageDataEntry): Button {
-    return pkg.installed?.['marketplace-url']
-      ? {
-          action: () =>
-            this.navCtrl.navigateForward([`marketplace/${pkg.manifest.id}`]),
-          title: 'Marketplace',
-          description: 'View service in marketplace',
-          icon: 'storefront-outline',
-        }
-      : {
-          disabled: true,
-          action: () => {},
-          title: 'Marketplace',
-          description:
-            'This package has been side-loaded and is not available in the Start9 Marketplace',
-          icon: 'storefront-outline',
-        }
+  private viewInMarketplaceButton(
+    pkg: PackageDataEntry,
+    currentMarketplace: Marketplace | null,
+  ): Button {
+    const pkgMarketplace = pkg.installed?.['marketplace-url']
+    const pkgM = pkgMarketplace
+      ? removeTrailingSlash(pkgMarketplace)
+      : pkgMarketplace
+    const currentM = currentMarketplace
+      ? removeTrailingSlash(currentMarketplace.url)
+      : currentMarketplace
+    if (!pkgMarketplace) {
+      return {
+        disabled: true,
+        action: () => {},
+        title: 'Marketplace',
+        description:
+          'This package has been sideloaded and is not available in the Start9 Marketplace',
+        icon: 'storefront-outline',
+      }
+    } else if (pkgM && currentM && pkgM !== currentM) {
+      return {
+        action: async () => {
+          const alert = await this.alertCtrl.create({
+            header: 'Marketplace Conflict',
+            message: `This service was installed from:
+            <br><br>
+            <span class="courier-new color-success-shade">${pkgM}</span>
```

Comment:
Maybe better to display the Marketplace names rather than URLs?

Reply:
We don't always have that information from the installed package details, it's just the marketplace url. We could attempt to use that value to look up the name from the saved marketplaces, but it's not guaranteed to exist there.